### ISSUE TITLE
Add support for setting an offset date/time for the RTC.

### ISSUE THREAD

**Issue by nl255** [yuzu-emu/yuzu#10862]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

Right now it is only possible to set a specific custom date for the RTC.  I would like to see a feature where you can set an offset (such as -60 days or +15 days) like you can in Citra that way you can just set it and forget it unless you actively want to "time travel".  I have attached an image of how Citra does it with the offset configuration highlighted in red.

![citra-offsetdate](https://github.com/yuzu-emu/yuzu/assets/4340587/e34484bd-9d76-4a20-9680-84f1cb45651b)

### Why would this feature be useful?

Right now you have to set a the date/time and keep updating it.  With an offset you can just set it and forget it.  While this is possible with software such as runasdate on Windows (and possibly faketime/libfaketime on Linux) it would be nice to have it built in.

**Comment by Squall-Leonhart** [comment]
Duplicate of https://github.com/yuzu-emu/yuzu/issues/10127?

**Comment by nl255** [comment]
> Duplicate of #10127?

No, that one is about the custom RTC not being effective unless you completely close yuzu and restart.  This is about allowing configuring an offset (i.e. system time +/- X days) rather than an absolute time and date.  I don't even see how you could think that it could be a duplicate of that one (unless you were just skimming for keywords).

**Comment by Squall-Leonhart** [comment]
Because it encompasses changes to the same feature and this won't work without that anyway and #10235 was already closed as dupe and IS the same issue as this one.
